php/libraries/FilesUploadHandler.class.inc

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * An interface containing methods for classes used to handle file
+ * uploads from a client to a server.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Handles file uploads from a client to a server. The methods listed in this
+ * interface were created in order to encapsulate the usual things that happen
+ * when a set of files is uploaded to a server, namely:
+ *
+ * 1. Validation of all uploaded files
+ * 2. Transfer of all files to a designated (permanent) location on the
+ *    server.
+ * 3. Actual processing of the uploaded files (e.g. update of database
+ *    tables, etc...)
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+interface FilesUploadHandler
+{
+    /**
+     * Returns the properties associated to each file uploaded on the
+     * server in an array identical to $_FILES.
+     *
+     * @return array the properties for each file uploaded (one array for
+     *         each uploaded file, see $_FILES).
+     */
+    public function getFiles() : array;
```

Comment:
I made this comment on the other PR, but I think an iterable of PSR7 UploadedFileInterface's is better than an array of _FILES (most of the other comments on this review build on reasons why), but in addition to that, I'm not sure if this is even required, since once the transition to a PSR7/15 based router is finished, this will essentially just be a duplicate of ServerRequestInterface->getUploadedFiles() in a different format.

What I think the interface itself needs is a way to inject *which* files it's handling (ie either an immutable `withFiles` or a mutable `setFiles`, my preference would be the former), rather than making assumptions about the execution environment.